Automattic/woocommerce-payments
# perf: optimize settings re-rendering

Fixes N/A

#### Changes proposed in this Pull Request









I noticed a little performance improvement we can make on the settings page.
When changing any of the settings, the whole React tree is re-rendering.
The performance can be improved by reducing the number of components being re-rendered when performing a change.

Before the changes (highlights are the components re-rendering):
![2021-06-10 13 34 59](https://user-images.githubusercontent.com/273592/121580669-a0160b80-c9f2-11eb-9043-07c583715e6b.gif)

After the changes:
![2021-06-10 13 35 42](https://user-images.githubusercontent.com/273592/121580699-a73d1980-c9f2-11eb-9b21-611380c4e524.gif)

As you can see, there are way fewer components re-rendering when any of the settings are being changed.

I measured this, and it seems like it would be about 50% faster, just by abstracting the loadable component into its own reusable thing.

Before the changes:
![Screen Shot 2021-06-10 at 1 47 10 PM](https://user-images.githubusercontent.com/273592/121580838-cf2c7d00-c9f2-11eb-96f7-cb4f8bad36d8.png)

After the changes:
![Screen Shot 2021-06-10 at 1 44 02 PM](https://user-images.githubusercontent.com/273592/121580849-d3589a80-c9f2-11eb-9f0f-89d3ec906b33.png)


#### Testing instructions





- In WCPay dev, ensure you have the "Enable grouped settings" feature enabled
- Go to the WCPay settings page: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Change some settings
- Functionality stays the same

If you want to, you can enable the drawing updates in the tools:
![Markup 2021-06-10 at 13 52 30](https://user-images.githubusercontent.com/273592/121581124-1f0b4400-c9f3-11eb-9915-9cad791c7cc2.png)

You can also measure this in the "performance" tab in Chrome. I used a 6x CPU slowdown.

-------------------
